File: tnkb/exceptions.py
"""
TNKB Client - Exception Classes
"""


class TNKBError(Exception):
    """Base exception for TNKB errors"""
    pass


class InvalidPlateError(TNKBError):
    """Raised when an invalid plate number is provided"""
    pass


class APIError(TNKBError):
    """Raised when API request fails"""
    pass


class NetworkError(APIError):
    """Raised when network connection fails"""
    pass


class TimeoutError(APIError):
    """Raised when API request times out"""
    pass


class ValidationError(TNKBError):
    """Raised when validation fails"""
    pass

File: php/src/Exceptions.php
<?php

declare(strict_types=1);

namespace TNKB;

/**
 * Base exception for all TNKB errors
 */
class TNKBException extends \Exception
{
}

/**
 * Thrown when plate number format is invalid
 */
class InvalidPlateException extends TNKBException
{
}

/**
 * Thrown when API returns an error
 */
class APIException extends TNKBException
{
}

/**
 * Thrown when network connection fails
 */
class NetworkException extends APIException
{
}

/**
 * Thrown when request times out
 */
class TimeoutException extends NetworkException
{
}

/**
 * Thrown when validation fails
 */
class ValidationException extends TNKBException
{
}
